Makefile

```diff
@@ -40,49 +44,65 @@ $(PYTHON3):
 
 
 #Install the project via setup.py
-.PHONEY: venv-install
+.PHONY: venv-install
 venv-install: $(CARAMEL_TOOL)
 $(CARAMEL_TOOL): $(PYTHON3) setup.py
 	@$(BOLD); echo "Install caramel and its dependencies in venv: $(VENV)";\
 	$(BLR);
 	$(VENV)/bin/pip3 install -e .
+	cp development.ini $(VENV)/development.ini
+	mkdir $(VENV)/example_ca
 	@$(BLR)
 
 # Create a sqlite-db configured for use with caramel
-$(DB_FILE): $(CARAMEL_TOOL)
+.PHONY: gen-db%
+gen-db%: $(CARAMEL_TOOL)
 	@$(BOLD); echo "Create a new DB at $(DB_FILE)";\
 	$(BLR);
-	$(VENV)/bin/caramel_initialize_db
+	$(VENV)/bin/caramel_initialize_db $(CARAMEL_COMMAND_LINE)
 	@$(BLR)
 
 # Generate a new CA cert and key pair
-$(CA_CERT): $(CARAMEL_TOOL)
+.PHONY: ca-cert%
+ca-cert%: $(CARAMEL_TOOL)
 	@$(BOLD); echo "Generate new CA cert and key with tests/ca_test_input.txt";\
 	$(BLR)
-	$(VENV)/bin/caramel_ca < tests/ca_test_input.txt
+	$(VENV)/bin/caramel_ca $(CARAMEL_COMMAND_LINE) < tests/ca_test_input.txt
 	@$(BLR)
 
-# Start caramel using pserve in the background, save PID to SERVER_PID
-$(SERVER_PID): $(CA_CERT) $(DB_FILE)
+# Start caramel using pserve in the background, save PID to SERVER
+$(SERVER)-env.pid: ca-cert-env gen-db-env
 	@ $(BOLD); echo "Start new caramel server in the background, sleep 2s to \
 	give it time to start";\
 	$(BLR)
 	chmod +x scripts/caramel_launcher.sh
 	setsid ./scripts/caramel_launcher.sh $(VENV)/bin/pserve >/dev/null 2>&1 < /dev/null & \
-	echo $$! > $(SERVER_PID)
+	echo $$! > $(SERVER)-env.pid
 	sleep 2s
 	@$(BLR)
 
-# Start caramel_autosign in the background, save PID to AUTOSIGN_PID
-$(AUTOSIGN_PID): $(CARAMEL_TOOL) $(CA_CERT) $(DB_FILE)
+
+# Start caramel using pserve in the background, save PID to SERVER
+$(SERVER)-in%.pid: ca-cert-in% gen-db-in%
+	@ $(BOLD); echo "Start new caramel server in the background, sleep 2s to \
+	give it time to start";\
+	$(BLR)
+	setsid $(VENV)/bin/pserve $(CARAMEL_COMMAND_LINE) >/dev/null 2>&1 < /dev/null & \
+	echo $$! > $(SERVER)-in$*.pid
```

Comment:
you probably want this line to be:
echo $$! > $@

`$@` is the fully expanded target.

There's a few other places above, like $(SERVER)-env.pid   and the `$(AUTOSIGN)$*.pid` below that should both be `$@` instead of re-expanding things.

It doesn't matter overly here, so treat this mostly as commentary.